packages/app/src/components/ErrorBoundaries/FatalErrorBoundary.js

```diff
@@ -0,0 +1,179 @@
+        textDecoration: 'underline',
+        cursor: 'pointer',
+        marginBottom: 12,
+    },
+    drawerVisible: {
+        padding: 8,
+        display: 'block',
+        height: 150,
+        width: 500,
+        overflow: 'auto',
+        overflowY: 'auto',
+        border: `1px solid ${secondaryTextColor}`,
+        textAlign: 'left',
+    },
+    drawerHidden: {
+        display: 'none',
+    },
+    errorIntro: {
+        fontSize: '12px',
+        lineHeight: 1.2,
+        color: secondaryTextColor,
+        marginBottom: 8,
+        fontFamily: 'Menlo, Courier, monospace !important',
+    },
+    errorDetails: {
+        fontSize: '12px',
+        lineHeight: 1.2,
+        color: 'red',
+        fontFamily: 'Menlo, Courier, monospace !important',
+    },
+};
+
+class FatalErrorBoundary extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            error: null,
+            errorInfo: null,
+            drawerOpen: false,
+        };
+    }
+
+    componentDidCatch(error, info) {
+        this.setState({
+            error,
+            errorInfo: info,
+        });
+    }
+
+    toggleTechInfoDrawer = () => {
+        this.setState({
+            drawerOpen: !this.state.drawerOpen,
+        });
+    };
+
+    render() {
+        const { classes, children } = this.props;
+        if (this.state.error) {
+            return (
+                <div className={classes.mask}>
+                    <div className={classes.container}>
+                        <InfoIcon className={classes.icon} />
+                        <div className={classes.message}>
+                            {i18n.t('Something went wrong')}
+                        </div>
+                        <div
+                            className={classes.link}
+                            onClick={() => window.location.reload()}
+                        >
+                            {i18n.t('Refresh to try again')}
+                        </div>
+                        <div
+                            className={classes.drawerToggle}
+                            onClick={this.toggleTechInfoDrawer}
+                        >
+                            {this.state.drawerOpen
+                                ? i18n.t('Hide technical details')
+                                : i18n.t('Show technical details')}
+                        </div>
+                        <div
+                            className={
+                                this.state.drawerOpen
+                                    ? classes.drawerVisible
+                                    : classes.drawerHidden
+                            }
+                        >
+                            <div className={classes.errorIntro}>
+                                {i18n.t(
+                                    'An error occurred in the DHIS2 {{AppName}} application.',
+                                    { AppName: 'Data Visualizer' }
```

Comment:
We do translate the app name in the headerbar, so perhaps we can remove the variable so that the whole string can be translated with the app name in it.